test/parser_test.exs

```diff
@@ -697,4 +699,43 @@ defmodule ParserTest do
     |> parse
   end
 
+  test "parsing a file with syntax errors" do
+    contents = """
+    stract Typo {
+        1: optional i32 id
+    }
+    """
+
+    assert {:error, _} = parse(contents)
+
+    dir = Path.join([@project_root, "tmp", "syntax_error_test"])
+    File.mkdir_p!(dir)
+    path = Path.join(dir, "syntax_error.thrift")
+    File.write!(path, contents)
+
+    assert_raise(
+      Thrift.FileParseException,
+      ~r/#{path} on line 1:/,
+      fn -> parse_file(path) end
+    )
+
+    other_path = Path.join(dir, "includes_syntax_error.thrift")
+    File.write!(other_path, """
+    include "syntax_error.thrift"
+
+    struct NoError {
+      1: optional i32 id
+    }
+    """)
+
+    # should raise an error on the included file,
+    # since that is where the syntax error is
+    assert_raise(
+      Thrift.FileParseException,
+      ~r/#{path} on line 1:/,
+      fn -> parse_file(other_path) end
+    )
+
+    File.rm_rf!(dir)
```

Comment:
True.  I didn't like the way this was set up, but wasn't sure where else to put it.  It seemed awkward to have a setup/teardown execute this for every test in the file where this is the only  test that actually uses that directory.  I could move it to its own file, I suppose?